core/errors/error-match-error.ts

```diff
@@ -2,7 +2,7 @@ import { MatchError } from "./match-error";
 
 export class ErrorMatchError extends MatchError {
 
-   public constructor(actualError: Error,
+   public constructor(actualError: Error | undefined | null,
```

Comment:
Sure, I'm thinking that it should be Error | null though and keep a consistency here in that it is nothing on purpose i.e. `null`?